instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -53,33 +53,31 @@
 from .version import __version__
 
 _logger = logging.getLogger(__name__)
-# We use this prefix so we can request all instrumentation MessageAttributeNames with a wildcard, without harming
-# existing filters
-_OPENTELEMETRY_ATTRIBUTE_IDENTIFIER: str = "otel."
-_OTEL_IDENTIFIER_LENGTH = len(_OPENTELEMETRY_ATTRIBUTE_IDENTIFIER)
+
+_IS_SQS_INSTRUMENTED_ATTRIBUTE = "_otel_boto3sqs_instrumented"
 
 
 class Boto3SQSGetter(Getter[CarrierT]):
     def get(self, carrier: CarrierT, key: str) -> Optional[List[str]]:
-        value = carrier.get(f"{_OPENTELEMETRY_ATTRIBUTE_IDENTIFIER}{key}", {})
-        if not value:
+        msg_attr = carrier.get(key)
+        if not isinstance(msg_attr, Mapping):
+            return None
+
+        value = msg_attr.get("StringValue")
+        if value is None:
             return None
-        return [value.get("StringValue")]
+
+        return [value]
 
     def keys(self, carrier: CarrierT) -> List[str]:
-        return [
-            key[_OTEL_IDENTIFIER_LENGTH:]
-            if key.startswith(_OPENTELEMETRY_ATTRIBUTE_IDENTIFIER)
-            else key
-            for key in carrier.keys()
-        ]
+        return list(carrier.keys())
 
 
 class Boto3SQSSetter(Setter[CarrierT]):
     def set(self, carrier: CarrierT, key: str, value: str) -> None:
         # This is a limitation defined by AWS for SQS MessageAttributes size
         if len(carrier.items()) < 10:
```

Comment:
AWS does not accept keys that start with `AWS` or `Amazon`. Can we add another check here that enforces that, and if this is the case a warning will be printed?

Reply:
I don't think guarding against this very specific case is worth it, unless you know any propagator that actually uses such keys? There are rules about periods/dots in the name as well and a maximum length which we also don't check.

Reply:
You're right. We should also enforce these. Otherwise, the behavior would be a failure to send SQS messages *because* the user added OpenTelemetry. Adding OpenTelemtry must not alter the original behavior of the application

Reply:
That's an ideal, not something we can always guarantee. Checking for startswith AWS/Amazon sounds at least reasonably easy & fast, but should we really run a regex over each request? It just feels like going too far to me.

Reply:
I think it's something that cannot be ignored. The instrumentation must comply with the amazon API

Reply:
We also don't check that for any HTTP instrumentation. There are rules for HTTP headers as well. So it surely _can_ be ignored 😃